### PR TITLE
2015 06 mn spd add userbadges

### PR DESCRIPTION
**Pull request by MagdaN**
I am going to wire up the ui from @local-girl as well, and then you can review

### CODE REVIEW

**Comment by MagdaN** [comment]
This can be reviewed now - if you need a database, let me know. The badge jsons need to changed due to changes in another branch, but I will do this in the other branch (https://github.com/liqd/adhocracy3.mercator/pull/1363).

File: src/adhocracy_frontend/adhocracy_frontend/static/js/Packages/User/Views.ts [hunk 1]
```diff
@@ -352,12 +358,36 @@ export var metaDirective = (adhConfig : AdhConfig.IService, adhResourceArea : Ad
...
+        return adhHttp.get(assignmentPath).then((assignment : RIBadgeAssignment) => {
+            var description = assignment.data[SIDescription.nick].description;
+            return adhHttp.get(assignment.data[SIBadgeAssignment.nick].badge).then((badge) => {
+                var title = badge.data[SITitle.nick].title;
+                var name = badge.data[SIName.nick].name;
+                return new BadgeAssignment(title, description, name);
+            });
+        });
+    }));
+};
```
Comment (xi) [inline src/adhocracy_frontend/adhocracy_frontend/static/js/Packages/User/Views.ts]: This code is duplicated in `MeinBerlin/Kiezkassen/Proposal/Proposal.ts` and in `MercatorProposal/MercatorProposal.ts`. Please move it to a single place.

File: src/adhocracy_frontend/adhocracy_frontend/static/js/Packages/User/Views.ts [hunk 2]
```diff
@@ -430,6 +464,9 @@ export var userProfileDirective = (
                 adhHttp.resolve(scope.path)
                     .then((res) => {
                         scope.userBasic = res.data[SIUserBasic.nick];
+                        getBadges(adhHttp, $q)(res).then((assignments) => {
+                            scope.assignments = assignments;
+                        });
```
Comment (xi) [inline src/adhocracy_frontend/adhocracy_frontend/static/js/Packages/User/Views.ts]: I think this should be named `badges`, not `assignments`. I think we should rename it here and also in the other places where `getBadges()` is used.

**Comment by xi** [comment]
The bades should also be shown in the `adhUserMeta` directive.

**Comment by xi** [comment]
Apart from the comments above this looks good.

File: src/adhocracy_frontend/adhocracy_frontend/static/js/Packages/User/Views.ts (hunk 1 above)
Comment (MagdaN) [inline src/adhocracy_frontend/adhocracy_frontend/static/js/Packages/User/Views.ts]: I would like to do that in another branch, as it effects all badges not just user badges

File: src/adhocracy_frontend/adhocracy_frontend/static/js/Packages/User/Views.ts (hunk 2 above)
Comment (MagdaN) [inline src/adhocracy_frontend/adhocracy_frontend/static/js/Packages/User/Views.ts]: I would like to do that in another branch, as it effects all badges not just user badges.

**Comment by xi** [comment]
LGTM.

**Pull request merged**